HackStudio: Perform save-as action if build called before file is saved

atharva1910 opened the pull request:
If Build action is called before the file is saved, it leads to Hack studio crashing. This commit calls the save-as flow before the build if file is not saved before

MacDue reviewed (commented):
Should the word "Build" be capitalized in the second commit?

Also maybe it's just me but I'm struggling to parse the commit message `HackStudio: Change check from to document for editor wrapper`.

atharva1910 commented:
Hi @MacDue, Thanks for the review. I had a major typo in the commit message. I have changed both the commit message to a more saner message and fixed the capital 'B'
Thanks
